src/PoolManager.sol

```diff
@@ -243,6 +253,8 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
         onlyByLocker
         returns (BalanceDelta delta)
     {
+        CurrentHookAddress.set(address(key.hooks));
```

Comment:
I do think we should be clearing this after each call? It feels wrong that at the end of a call to `swap`, the control passes back to the locker, and `CurrentHookAddress` is still set to some hook that is _not_ the current hook?

This contract does pass control out to other contracts at times (like the locker, tokens), so i just want to lower the risk of when contracts can take control of the lock again?